packages/eservice-template-process/src/routers/EServiceTemplateRouter.ts

```diff
@@ -0,0 +1,754 @@
+  }),
+  readModelService,
+  initFileManager(config)
+);
+
+const eserviceTemplatesRouter = (
+  ctx: ZodiosContext
+): ZodiosRouter<ZodiosEndpointDefinitions, ExpressContext> => {
+  const {
+    ADMIN_ROLE,
+    API_ROLE,
+    SECURITY_ROLE,
+    M2M_ROLE,
+    SUPPORT_ROLE,
+    INTERNAL_ROLE,
+  } = userRoles;
+
+  return ctx
+    .router(eserviceTemplateApi.processApi.api, {
+      validationErrorHandler: zodiosValidationErrorToApiProblem,
+    })
+    .get(
+      "/eservices/templates",
+      authorizationMiddleware([
+        ADMIN_ROLE,
+        API_ROLE,
+        SECURITY_ROLE,
+        M2M_ROLE,
+        SUPPORT_ROLE,
+      ]),
+      async (req, res) => {
+        const ctx = fromAppContext(req.ctx);
+
+        try {
+          const {
+            name,
+            creatorsIds,
+            eserviceTemplatesIds,
+            states,
+            offset,
+            limit,
+          } = req.query;
+
+          const eserviceTemplates =
+            await eserviceTemplateService.getEServiceTemplates(
+              {
+                eserviceTemplatesIds:
+                  eserviceTemplatesIds.map<EServiceTemplateId>(unsafeBrandId),
+                creatorsIds: creatorsIds.map<TenantId>(unsafeBrandId),
+                states: states.map(
+                  apiEServiceTemplateVersionStateToEServiceTemplateVersionState
+                ),
+                name,
+              },
+              offset,
+              limit,
+              ctx
+            );
+
+          return res.status(200).send(
+            eserviceTemplateApi.EServiceTemplates.parse({
+              results: eserviceTemplates.results.map(
+                eserviceTemplateToApiEServiceTemplate
+              ),
+              totalCount: eserviceTemplates.totalCount,
+            })
+          );
+        } catch (error) {
+          return res.status(500).send();
+        }
+      }
+    )
+    .post(
+      "/eservices/templates",
+      authorizationMiddleware([ADMIN_ROLE, API_ROLE]),
+      async (req, res) => {
+        const ctx = fromAppContext(req.ctx);
+
+        try {
+          const eserviceTemplate =
+            await eserviceTemplateService.createEServiceTemplate(req.body, ctx);
+          return res
+            .send(
+              eserviceTemplateApi.EServiceTemplate.parse(
+                eserviceTemplateToApiEServiceTemplate(eserviceTemplate)
+              )
+            )
+            .status(200);
+        } catch (error) {
+          const errorRes = makeApiProblem(
+            error,
+            createEServiceTemplateErrorMapper,
+            ctx.logger,
+            ctx.correlationId
+          );
+          return res.status(errorRes.status).send(errorRes);
+        }
+      }
+    )
+    .get(
+      "/eservices/templates/:eServiceTemplateId",
+      authorizationMiddleware([
+        ADMIN_ROLE,
+        API_ROLE,
+        SECURITY_ROLE,
+        M2M_ROLE,
+        SUPPORT_ROLE,
+        INTERNAL_ROLE,
+      ]),
```

Comment:
Internal role is not needed here I think

```suggestion
      authorizationMiddleware([
        ADMIN_ROLE,
        API_ROLE,
        SECURITY_ROLE,
        M2M_ROLE,
        SUPPORT_ROLE,
      ]),
```

Reply:
I think it's needed, this route is called by the `bff`

Reply:
BFF and API GW do not use internal tokens: they forward the token coming from the caller of the API, which is an Interop token with one of the user roles (Admin, etc.).
Internal is needed only for the cases where the API is not called by a tenant, cases where an internal token is generated by another service in the platform (e.g., consumer services reacting to some events).
In fact, internal tokens don't have an organizationId and any call that uses authData.organizationId in its logic will fail at runtime if called with an Internal token

Reply:
Didn't know that! Thank you for taking the time explaining it 🙏